File: Software/Chapter 15/cs/Listing15-2/Listing15-2/Listing15-2.cs
using System;
using System.Data;
using System.Data.SqlClient;

namespace APress.Samples
{
    class DataReaderExample
    {
        static void Main(string[] args)
        {
            String sqlConStr = "DATA SOURCE=(local);" +
                "INITIAL CATALOG=AdventureWorks;" +
                "INTEGRATED SECURITY=SSPI;";
            String sqlStmt = "SELECT DepartmentId, " +
                "     Name, " +
                "     GroupName, " +
                "     ModifiedDate " +
                " FROM HumanResources.Department " +
                " ORDER BY DepartmentId";
            SqlDataAdapter sqlDa = null;
            DataSet ds = null;
            try
            {
                sqlDa = new SqlDataAdapter(sqlStmt, sqlConStr);
                ds = new DataSet();
                sqlDa.Fill(ds);
                foreach (DataRow dr in ds.Tables[0].Rows)
                {
                    Console.WriteLine("{0}\t{1}\t{2}\t{3}",
                        dr["DepartmentId"].ToString(), 
                        dr["Name"].ToString(), 
                        dr["GroupName"].ToString(), 
                        dr["ModifiedDate"].ToString());
                }
            }
            catch (SqlException ex)
            {
                Console.WriteLine(ex.Message);
            }
            finally
            {
                if (ds != null)
                {
                    ds.Dispose();
                }
                if (sqlDa != null)
                {
                    sqlDa.Dispose();
                }
            }
            Console.Write("Press a Key to Continue...");
            Console.ReadKey();
        }
    }
}

File: Software/Chapter 15/vb/Listing15-2/Listing15-2/Listing15-2.vb
Imports System.Data
Imports System.Data.SqlClient

Namespace APress.Samples
    Module DataReaderExample
        Sub Main()
            Dim sqlConStr As String = "DATA SOURCE=(local);" & _
                "INITIAL CATALOG=AdventureWorks;" & _
                "INTEGRATED SECURITY=SSPI;"
            Dim sqlStmt As String = "SELECT DepartmentId, " & _
                "     Name, " & _
                "     GroupName, " & _
                "     ModifiedDate " & _
                " FROM HumanResources.Department " & _
                " ORDER BY DepartmentId"
            Dim sqlDa As SqlClient.SqlDataAdapter = Nothing
            Dim ds As DataSet = Nothing
            Try
                sqlDa = New SqlClient.SqlDataAdapter(sqlStmt, sqlConStr)
                ds = New DataSet
                sqlDa.Fill(ds)
                For Each dr As DataRow In ds.Tables(0).Rows
                    Console.WriteLine("{0}" & ControlChars.Tab & _
                        "{1}" & ControlChars.Tab & _
                        "{2}" & ControlChars.Tab & _
                        "{3}", dr.Item("DepartmentId").ToString(), _
                        dr.Item("Name").ToString(), _
                        dr.Item("GroupName").ToString(), _
                        dr.Item("ModifiedDate").ToString())
                Next
            Catch ex As SqlException
                Console.WriteLine(ex.Message)
            Finally
                If Not (ds Is Nothing) Then
                    ds.Dispose()
                End If
                If Not (sqlDa Is Nothing) Then
                    sqlDa.Dispose()
                End If
            End Try
            Console.Write("Press a Key to Continue...")
            Console.ReadKey()
        End Sub
    End Module
End Namespace

